pkg/proxy/proxy.go

```diff
@@ -143,10 +143,21 @@ func (n krpAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *http
 	allAttrs := []authorizer.Attributes{}
 
 	if n.authzConfig.ResourceAttributes != nil {
-		if n.authzConfig.Rewrites != nil && n.authzConfig.Rewrites.ByQueryParameter != nil && n.authzConfig.Rewrites.ByQueryParameter.Name != "" {
-			params, ok := r.URL.Query()[n.authzConfig.Rewrites.ByQueryParameter.Name]
-			if !ok {
-				return nil
+		if n.authzConfig.Rewrites != nil {
+			params := []string{}
+			if n.authzConfig.Rewrites.ByQueryParameter != nil && n.authzConfig.Rewrites.ByQueryParameter.Name != "" {
+				if ps, ok := r.URL.Query()[n.authzConfig.Rewrites.ByQueryParameter.Name]; ok {
+					params = append(params, ps...)
+				}
+			}
+			if n.authzConfig.Rewrites.ByHTTPHeader != nil && n.authzConfig.Rewrites.ByHTTPHeader.Name != "" {
+				if p := r.Header.Get(n.authzConfig.Rewrites.ByHTTPHeader.Name); p != "" {
+					params = append(params, p)
+				}
+			}
+
+			if len(params) == 0 {
+				return []authorizer.Attributes{}
```

Comment:
nit: i would continue returning `nil` here, there is no need to allocate an empty slice, as the zero value (`nil`) of a slice also returns 0 for its capacity and length.